server/api/region.go

```diff
@@ -726,6 +728,56 @@ func (h *regionsHandler) ScatterRegions(w http.ResponseWriter, r *http.Request)
 	h.rd.JSON(w, http.StatusOK, &s)
 }
 
+// @Tags region
+// @Summary Split regions with given split keys
+// @Accept json
+// @Param body body object true "json params"
+// @Produce json
+// @Success 200 {string} string "Split regions with given split keys"
+// @Failure 400 {string} string "The input is invalid."
+// @Router /regions/split [post]
+func (h *regionsHandler) SplitRegions(w http.ResponseWriter, r *http.Request) {
+	rc := getCluster(r.Context())
+	var input map[string]interface{}
+	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &input); err != nil {
+		return
+	}
+	rawSplitKeys, ok := input["split_keys"].([]interface{})
+	if !ok {
+		h.rd.JSON(w, http.StatusBadRequest, "split_keys should be provided.")
+		return
+	}
+	if len(rawSplitKeys) < 1 {
+		h.rd.JSON(w, http.StatusBadRequest, "empty split keys.")
+		return
+	}
+	retryLimit, ok := input["retry_limit"].(int)
+	if !ok {
+		retryLimit = 5
+	}
+	splitKeys := make([][]byte, 0, len(rawSplitKeys))
+	for _, rawKey := range rawSplitKeys {
+		key, err := hex.DecodeString(rawKey.(string))
+		if err != nil {
+			h.rd.JSON(w, http.StatusBadRequest, err.Error())
+			return
+		}
+		splitKeys = append(splitKeys, key)
+	}
+	s := struct {
+		ProcessedPercentage int      `json:"processed-percentage"`
+		NewRegionsID        []uint64 `json:"regions-id"`
+	}{}
+	percentage, newRegionsID := rc.GetRegionSplitter().SplitRegions(splitKeys, retryLimit)
+	s.ProcessedPercentage = percentage
+	s.NewRegionsID = newRegionsID
+	failpoint.Inject("splitResponses", func() {
+		s.ProcessedPercentage = 100
+		s.NewRegionsID = []uint64{99, 100, 101}
```

Comment:
the values look test case specific. I think it would be better to use [failpoint.Value](https://github.com/pingcap/failpoint#how-to-inject-a-failpoint-to-your-program) to inject from test case.

Reply:
updated.